app/views/layouts/darkswarm.html.haml

```diff
@@ -48,8 +48,10 @@
     = inject_current_hub
     = inject_current_user
     = inject_rails_flash
-    = inject_taxons
-    = inject_properties
+    - cache "inject-all-taxons-#{CacheService.latest_timestamp_by_class(Spree::Taxon)}" do
+      = inject_taxons
+    - cache "inject-all-properties-#{CacheService.latest_timestamp_by_class(Spree::Property)}" do
+      = inject_properties
```

Comment:
This is really great! We're caching two big AMS arrays (all the taxons and properties in the database) _after_ they're fetched, serialized, and rendered, and expiring the cache entries automatically. We'll be saving a lot of CPU time here on every page load.

Reply:
:+1:
we need an auto test that covers the case where we just assign an existing property to a product and that change is reflected in the cached value. what are all possible ways of assigning properties to products? I think it's in the product properties page.
The same applies to taxons. I dont think we have a way to add taxons to a product!? Only changing the category of the product (primary taxon).

Reply:
In this specific cases in this view, the objects being rendered here are literally just `Spree::Taxon.all` and `Spree::Property.all`, with no absolutely relation to whether or not they're applied to any products, or producers, or anything. And their serializers are not calling associated data.

But yes, we'll need to do that in the queries being cached in `EnterpriseInjectionData`, which do involve some joins and associations and need much more careful handling :+1:

Reply:
ok :+1: You still need to test for the create taxon case and create properties.
You can create properties from both the properties page and the product properties page.

Reply:
Saving a new record always sets it's `:updated_at` value to the current time, right?

Reply:
I've removed some of the caching on one of the more complicated queries, updated the touches and touch specs and added some code comments.